lib/katello/repository_types/README.md

```diff
@@ -28,24 +60,45 @@ Katello::RepositoryTypeManager.register(::Katello::Repository::YUM_TYPE) do
 ```
 This declares the the repository type, specifying a type identifier (e.g. 'yum')
 
-### Service class
+### Service class and friends
 
 ```ruby
-  service_class Katello::Pulp::Repository::Yum
+  pulp3_service_class Katello::Pulp3::Repository::Yum
+  pulp3_api_class Katello::Pulp3::Api::Yum
+  pulp3_plugin 'rpm'
+  ...
 ```
 
-Identifies the service class that is used to interface with pulp 2 and includes methods for creating and updating the type of repository
+Identifies the service class that is used to interface with Pulp and includes methods for creating and updating the type of repository. Also notes the location of the main API class and the name of the corresponding Pulp plugin.
+There are a number of other classes that need to be defined, so follow the example above to ensure that the new type has
+all the neccessary information.
  
  ### Content Type
  
 ```ruby
-  content_type Katello::Rpm, :priority => 1, :pulp2_service_class => ::Katello::Pulp::Rpm
+  default_managed_content_type Katello::Rpm::CONTENT_TYPE
+  content_type Katello::Rpm,
+    :priority => 1,
+    :pulp3_service_class => ::Katello::Pulp3::Rpm,
+    :primary_content => true,
+    :removable => true,
+    :uploadable => true
 ```
 
-Defines a content type to be used by the repository.  The first argument is the Active Record model class representing the content type.  This should extend the `Katello::Concerns::PulpDatabaseUnit` class.
+Firstly, default_managed_content_type sets the Rpm content type as the "default". The default managed type
+will be the type acted upon in places where it is not clear which content unit should be acted on. For example, during yum uploads,
+it will be assumed that the upload is of type Rpm unless otherwise specified.
+
+The content_type method defines a content type to be used by the repository.  The first argument is the Active Record model class representing the content type.  This should extend the `Katello::Concerns::PulpDatabaseUnit` class.
 The second argument is an options hash that includes:
 * priority - The ordering used when indexing or performing other operations (1 is highest, defaults to 99)
-* pulp2_service_class - The service class representing this content type in pulp 2
+* pulp3_service_class - The service class representing this content type in Pulp 3
+* primary_content - The indexing of this content type is recorded for display after syncing
+* removable - Content units of this type can be deleted from a repository
+* uploadable - Content units of this type can be uploaded to a repository
+* index - Whether or not this content units of the type should be indexed
+* index_on_pulp3 - Deprecated, was previously used when Pulp 2 and Pulp 3 lived together
```

Comment:
Should `index_on_pulp3` be removed as part of this PR, then? As of this review it is still used in `lib/katello/repository_types/yum.rb` and maybe some others (I didn't check)

Then we could also remove the reference to it here